RecoLocalCalo/HcalRecProducers/src/HcalReconstructor.cc

```diff
@@ -0,0 +1,118 @@
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "DataFormats/Common/interface/Handle.h"
+#include "DataFormats/HcalRecHit/interface/HcalRecHitCollections.h"
+#include "HeterogeneousCore/SonicTriton/interface/TritonClient.h"
+#include "HeterogeneousCore/SonicCore/interface/SonicEDProducer.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "DataFormats/HcalRecHit/interface/HBHERecHit.h"
+
+class HcalReconstructor : public SonicEDProducer<TritonClient>
+{
+public:
+    explicit HcalReconstructor(edm::ParameterSet const& cfg) : 
+        SonicEDProducer<TritonClient>(cfg),
+        fChannelInfoName_(cfg.getParameter<edm::InputTag>("ChannelInfoName")),
+        fTokChannelInfo_(this->consumes<HBHEChannelInfoCollection>(fChannelInfoName_))
+     
+    {
+        this->produces<HBHERecHitCollection>();
+        this->setDebugName("HcalReconstructor");
+    }
+
+    void acquire(edm::Event const& iEvent, edm::EventSetup const& iSetup, Input& iInput) override {
+
+	const HBHEChannelInfoCollection *channelInfo   = 0;
+	edm::Handle<HBHEChannelInfoCollection> hChannelInfo;
+	iEvent.getByToken(fTokChannelInfo_, hChannelInfo); 
+	channelInfo = hChannelInfo.product();
+
+	auto& input1 = iInput.begin()->second;
+	auto data1 = std::make_shared<TritonInput<float>>();
+	data1->reserve(std::distance(channelInfo->begin(), channelInfo->end()));
+
+	hcalIds_.clear(); 
+	
+	for(HBHEChannelInfoCollection::const_iterator itC = channelInfo->begin(); itC != channelInfo->end(); itC++){
+
+	    std::vector<float> input;
+	    const HBHEChannelInfo& pChannel(*itC);
+  	    const HcalDetId        pDetId = pChannel.id();
+ 	    hcalIds_.push_back(pDetId);    
+
+	    //FACILE uses iphi as a continuous variable
+	    input.push_back((float)pDetId.iphi());
+            input.push_back((float)pChannel.tsGain(0.));
+	    for (unsigned int itTS=0; itTS < pChannel.nSamples(); ++itTS) {
+		input.push_back((float)pChannel.tsRawCharge(itTS));
+	    }
+  
+	    //FACILE considers 7 Hcal depths as binary variables
+            for (int itDepth=1; itDepth < 8; itDepth++){
+		if (pDetId.depth() == itDepth)  input.push_back(1.f);
+		else				input.push_back(0.f);
+	    }
+
+	    //ieta is also encoded as a binary variable
+	    for (int itIeta = 0; itIeta < 30; itIeta++){
+		if (std::abs(pDetId.ieta()) == itIeta)  input.push_back(1.f);
+		else					input.push_back(0.f);
+	    }
+
+	    data1->push_back(input);
+	}
+
+	//set batch at maximum RHcollsize; pad the remainder
```

Comment:
it's not necessary to pad the remainder to fill up the max batch size
instead, count up how many hits you actually have in each event, and then call `client_.setBatchSize(n)`

Reply:
Replaced with suggested `client_.setBatchSize(channelInfo->size())`